propagator/opentelemetry-propagator-ot-trace/src/opentelemetry/propagators/ot_trace/__init__.py

```diff
@@ -0,0 +1,177 @@
+    Getter,
+    Setter,
+    TextMapPropagator,
+    TextMapPropagatorT,
+)
+
+OT_TRACE_ID_HEADER = "ot-tracer-traceid"
+OT_SPAN_ID_HEADER = "ot-tracer-spanid"
+OT_SAMPLED_HEADER = "ot-tracer-sampled"
+OT_BAGGAGE_PREFIX = "ot-baggage-"
+
+_valid_header_name = re_compile(r"^[\w_^`!#$%&'*+.|~]+$")
+_valid_header_value = re_compile(r"^[\t\x20-\x7e\x80-\xff]+$")
+_valid_extract_traceid = re_compile(r"[0-9a-f]{32}|[0-9a-f]{16}")
+_valid_extract_spanid = re_compile(r"[0-9a-f]{16}")
+_valid_inject_traceid = re_compile(r"[0-9a-f]{1,32}|[0-9a-f]{1,16}")
+_valid_inject_spanid = re_compile(r"[0-9a-f]{1,16}")
+
+
+class OTTracePropagator(TextMapPropagator):
+    """Propagator for the OTTrace HTTP header format"""
+
+    def extract(
+        self,
+        getter: Getter[TextMapPropagatorT],
+        carrier: TextMapPropagatorT,
+        context: Optional[Context] = None,
+    ) -> Context:
+
+        traceid = extract_first_element(
+            getter.get(carrier, OT_TRACE_ID_HEADER)
+        )
+
+        spanid = extract_first_element(getter.get(carrier, OT_SPAN_ID_HEADER))
+
+        sampled = extract_first_element(getter.get(carrier, OT_SAMPLED_HEADER))
+
+        if sampled == "true":
+            traceflags = TraceFlags.SAMPLED
+        else:
+            traceflags = TraceFlags.DEFAULT
+
+        if (
+            traceid != INVALID_TRACE_ID
+            and _valid_extract_traceid.match(traceid) is not None
+            and spanid != INVALID_SPAN_ID
+            and _valid_extract_spanid.match(spanid) is not None
+        ):
+            context = set_span_in_context(
+                DefaultSpan(
+                    SpanContext(
+                        trace_id=int(traceid, 16),
+                        span_id=int(spanid, 16),
+                        is_remote=True,
+                        trace_flags=traceflags,
+                    )
+                ),
+                context,
+            )
+
+            baggage = get_all(context) or {}
+
+            for key in getter.keys(carrier):
+
+                if not key.startswith(OT_BAGGAGE_PREFIX):
+                    continue
+
+                baggage[key[len(OT_BAGGAGE_PREFIX) :]] = extract_first_element(
+                    getter.get(carrier, key)
+                )
+
+            for key, value in baggage.items():
+                context = set_baggage(key, value, context)
+
+        return context
+
+    def inject(
+        self,
+        set_in_carrier: Setter[TextMapPropagatorT],
+        carrier: TextMapPropagatorT,
+        context: Optional[Context] = None,
+    ) -> None:
+
+        span_context = get_current_span(context).get_span_context()
+
+        if (
+            span_context.trace_id == INVALID_TRACE_ID
+            or _valid_inject_traceid.match(hex(span_context.trace_id)[2:])
+            is None
+            or span_context.span_id == INVALID_SPAN_ID
+            or _valid_inject_spanid.match(hex(span_context.span_id)[2:])
+            is None
```

Comment:
I still don't think we need any of these checks besides L116. The span context passed here should always be valid non-zero 128 bit integer (which can always be converted to hex and match the regex) OR zero (`INVALID_TRACE_ID`)

Reply:
Right, removed.